### PR TITLE
Add Clash config parser

### LINKED ISSUE #14: feat(parser): added clash config parser
A clash config parser should be added to parse remote clash config (eg. https://github.com/go4sharing/sub)

### PR DESCRIPTION
**Pull request by kimjune01**
Implements a Clash YAML configuration parser to extract proxy URLs from remote Clash configs, addressing the need to parse configs from sources like github.com/go4sharing/sub and github.com/ripaojiedian/freenode.

## What Changed

Added `FromClash` transformer that:
- Parses Clash YAML configs and extracts proxy entries
- Supports http, https, socks4, socks5, and shadowsocks proxy types
- Validates port ranges (1-65535) and handles multiple port formats (int, float64, string)
- Includes 10MB size limit to prevent OOM attacks
- Skips complex protocols (vmess, vless, trojan) that require complete parameter sets

Comprehensive test coverage (18 test cases) including edge cases:
- Invalid YAML
- Missing required fields
- Port validation
- Oversized inputs
- Mixed valid/invalid entries

Fixes #14

## Summary by Sourcery

Add a Clash YAML configuration transformer that extracts proxy URLs into a normalized text format and wires it into the existing transformer registry.

New Features:
- Introduce a FromClash transformer to parse Clash YAML configs and emit http, https, socks4, socks5, and Shadowsocks proxy URLs.

Build:
- Promote gopkg.in/yaml.v3 to a direct dependency for YAML parsing.

Tests:
- Add unit tests for the Clash transformer covering valid configs, missing fields, invalid YAML, port validation, protocol filtering, and oversized input handling.

## Summary by CodeRabbit

* **New Features**
  * Added a Clash proxy transformer that extracts proxy URLs for HTTP, HTTPS, SOCKS4/5, Shadowsocks, Vmess, Vless and Trojan, with transport (WS/GRPC/H2) and TLS/Reality options.

* **Bug Fixes / Validation**
  * Enforced port parsing and 1–65535 range, server-field validation, and a 10MB YAML input size limit; malformed or unsupported entries are skipped.

* **Tests**
  * Added comprehensive tests covering protocols, encoding, edge cases, validation, mixed entries, and oversized payloads.

* **Dependencies**
  * Promoted YAML library to a direct dependency.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gfpcom/free-proxy-list/pull/49)

### CODE REVIEW

**Comment by gfpcom** [comment]
@kimjune01  Good job 👍 .  Could you get some time left to verify the review from sourcery-ai? thanks.

File: internal/transformer.go [hunk 1]
```diff
@@ -36,3 +43,101 @@ func FromBase64(buf []byte) []byte {
...
+		}
+		password, ok := proxy["password"].(string)
+		if !ok || password == "" {
+			return ""
+		}
+		// Shadowsocks URL format: ss://base64(cipher:password)@server:port
+		credentials := base64.URLEncoding.EncodeToString([]byte(cipher + ":" + password))
+		return fmt.Sprintf("ss://%s@%s:%d", credentials, server, portInt)
+	case "vmess", "vless", "trojan":
+		// These protocols require complex URL formats with UUIDs, encryption params, etc.
```
Comment (gfpcom) [inline internal/transformer.go]: In general, almost of proxies from a clash.yml are vmess/vless. could you have a look at https://github.com/cnlangzi/proxyclient/tree/main/xray? VlessURL/VmessURL/TrojanURL/SSURL/SSRURL are implemented by him.

**Review by gfpcom — COMMENTED** [review]
@kimjune01  Excellent PR 👍 . It is appreciated if you can get some time left to improve it.

**Comment by kimjune01** [comment]
- Refactored to typed structs per your feedback
- Added vmess/vless/trojan builders following the proxyclient URL format you linked
- Auth credentials now included in HTTP/SOCKS URLs via `url.UserPassword()`
- IPv6 hosts bracketed via `net.JoinHostPort`
- Non-integer float ports rejected instead of silently truncated
- 48 test cases covering all protocol types

AI was used to assist with implementation. I reviewed and understand the change.

**Comment by gfpcom** [comment]
@kimjune01  thanks for your job. there are only two small issues found by coderabbitai to fix before getting it merged. 👍

**Comment by kimjune01** [comment]
Pushed both fixes 👍

- Float port casting in `FlexPort.UnmarshalYAML` now explicitly rejects NaN, Inf, non-integer floats, and out-of-range values before the `int()` cast (commit `215b2080`).
- Renamed the 10MB / oversized test cases to describe which gate they exercise (parser vs pre-parse size gate) with comments (commit `d9ea665f`).
- New `TestFlexPortFloatRejection` table test locks the contract: 8080 and 8080.0 accepted; 8080.5, -1.0, 99999.0, NaN, ±Inf, 0, 65536 rejected (commit `55d0804a`).

`go test ./...` passes locally.

**Comment by gfpcom** [comment]
@kimjune01 perfect 💯
